src/function_classes/linear.py

```diff
@@ -33,7 +32,7 @@ def __init__(self, sparsity: int = 3, *args, **kwargs):
         self._sparsity = sparsity
 
     def evaluate(self, x_batch: torch.Tensor, *params: torch.Tensor) -> torch.Tensor:
-        weights, _ = params
+        weights = torch.stack(params)
```

Comment:
here, `params` is of length 32, so the weights are precisely params, so calling `torch.stack` correctly gets the weights.